ops/model.py

```diff
@@ -230,6 +241,96 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping:
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, backend):
+        self._backend = backend
+        self._data = {}
+
+    def get(self, binding_name):
+        if isinstance(binding_name, Relation):
+            key = binding_name
+            name = binding_name.name
+            relation_id = binding_name.id
+        elif isinstance(binding_name, str):
+            name = key = binding_name
+            relation_id = None
+        else:
+            raise ModelError(f'binding_name must be a str or Relation, not {type(binding_name).__name__}')
+        binding = self._data.get(key)
+        if binding is None:
+            binding = Binding(name, relation_id, self._backend)
+            self._data[key] = binding
+        return binding
+
+
+class Binding:
+    """A representation of a binding to a space."""
+
+    def __init__(self, name, relation_id, backend):
+        self.name = name
+        self._relation_id = relation_id
+        self._backend = backend
+        self._network = None
+
+    @property
+    def network(self):
+        if self._network is None:
+            try:
+                if self._relation_id is None:
+                    network_info = self._backend.network_get(self.name)
+                else:
+                    network_info = self._backend.network_get(self.name, self._relation_id)
+            except RelationNotFoundError:
+                # If a relation is dead, we can still get network info associated with an endpoint itself
+                # not associated with a particular relation id.
+                network_info = self._backend.network_get(self.name)
+            self._network = Network(network_info['bind-addresses'], network_info['ingress-addresses'], network_info['egress-subnets'])
```

Comment:
Why are we splitting it up here if we're passing that data to the object that precisely is the one that knows about these attributes, and also knows about the mapping itself given it's digging further into it?

Reply:
why isn't `network_get` building the Network?

Reply:
... because i'ts in _backend which knows nothing of the higher level stuff. OK.

Reply:
@dshcherb to be clear (sorry if I confused things), the suggestion here is to do `Network(network_info)` and have Network fish things out.

Reply:
Good point, there is no need for Binding to dig into `network_info`.

Reply:
https://github.com/canonical/operator/pull/109/commits/10eeb01cf71dceaa69132d457a0f34eef09a6c2a